packages/core/src/modules/connections/protocols/trust-ping/v2/handlers/TrustPingMessageHandler.ts

```diff
@@ -16,11 +18,27 @@ export class TrustPingMessageHandler implements MessageHandler {
   }
 
   public async handle(messageContext: MessageHandlerInboundMessage<TrustPingMessageHandler>) {
+    let connection: ConnectionRecord | undefined = messageContext.connection
+
+    // If we received a thrust ping for nonexisting connection record,
+    // create a connection record when the corresponding option is set in the config
+    const recipient = messageContext.message.firstRecipient
```

Comment:
Are we always the first recipient? Should we update the messageContext to include which recipent did we are? I think this could lead to issues...

If we don't want to add this complexity now, I think we should throw an error if there's more than one recipient, so it couldn't lead to issues

Reply:
Added check that there is no more than one recipient for now.